### PR TITLE
Get CanvasHack working for import of a single http (basic) endpoint

### PR DESCRIPTION
**Pull request by StachuDotNet**
There's some cleanup that can be done here, but this gets the basic job done

### CODE REVIEW

File: backend/src/CanvasHack/AddOp.fs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+open Tablecloth
+
+module C = LibBackend.Canvas
+module Serialize = LibBackend.Serialize
+module Op = LibBackend.Op
+module PT = LibExecution.ProgramTypes
+
+let causesAnyChanges (ops : PT.Oplist) : bool = List.any Op.hasEffect ops
+
+let addOp
```
Comment (pbiggar) [inline backend/src/CanvasHack/AddOp.fs:17]: This does a lot more than I'd expect. Is it all necessary?

Comment (StachuDotNet) [reply backend/src/CanvasHack/AddOp.fs:17]: No, not necessary. I stole/adapted the code from the old AddOp ApiServer endpoint - we could rip out unnecessary parts if we need to, but didn't bother as I didn't fully understand the code and figured the whole project would be ripped out eventually. (I started down a path of reducing the logic here but broke it a few times, so stepped away)

Comment (StachuDotNet) [reply backend/src/CanvasHack/AddOp.fs:17]: Looking again now in the daylight, there are actually lots of trivial bits here that can be removed. I was just getting stumped trying to reduce the logic at the end.
So, I can tidy up here a bit.

Comment (StachuDotNet) [reply backend/src/CanvasHack/AddOp.fs:17]: Actually, yeah there's a ton to remove here - probably OK to just reframe the whole module as "addOpsFresh" and kill a bunch in here.

**Pull request merged**